ci/scripts/cpp_build.sh

```diff
@@ -283,9 +283,10 @@ else
 fi
 
 if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
-  time meson install
+  time meson compile -j ${ARROW_BUILD_PARALLEL:-$[${n_jobs} + 1]}
```

Comment:
Could you define `ARROW_BUILD_PARALLEL` out of this `if` to avoid the default value duplication?